src/modules/ekf2/EKF/ekf.h

```diff
@@ -802,13 +807,20 @@ class Ekf final : public EstimatorInterface
 
 
 	void resetHorizontalPositionToLastKnown();
+	void resetHorizontalPositionToExternal(const Vector2f &new_horiz_pos, float horiz_accuracy);
 
 	void resetHorizontalPositionTo(const Vector2f &new_horz_pos, const Vector2f &new_horz_pos_var);
 	void resetHorizontalPositionTo(const Vector2f &new_horz_pos, const float pos_var = NAN) { resetHorizontalPositionTo(new_horz_pos, Vector2f(pos_var, pos_var)); }
 
 	bool isHeightResetRequired() const;
 
 	void resetVerticalPositionTo(float new_vert_pos, float new_vert_pos_var = NAN);
+	template<typename T>
+	void recordResetStateChange(T &status_pos_change, const T &pos_diff, uint8_t &reset_count, uint8_t &reset_count_prev)
```

Comment:
To be honest I think using this is more confusing than doing it manually, you have to know both current and prev reset counts are tracked and the internal details.

Shorter term how about starting with more specific helpers?

``` C++
    void recordHorizontalPositionReset(const Vector2f &pos_diff);
```

I'm all for making things generic if we can, but the first priority should be easy to use, hard to misuse.